File: julia-package/brainflow/test/brainflow_get_data_twice.jl
using BrainFlow

# specify logging library to use
BrainFlow.enable_dev_logger(BrainFlow.BOARD_CONTROLLER)

params = BrainFlowInputParams()
board_shim = BrainFlow.BoardShim(BrainFlow.SYNTHETIC_BOARD, params)

BrainFlow.prepare_session(board_shim)
BrainFlow.start_stream(board_shim)
sleep(5)
BrainFlow.stop_stream(board_shim)
data = BrainFlow.get_board_data(board_shim)
println(data)
data = BrainFlow.get_board_data(10, board_shim)
println(data)
data = BrainFlow.get_current_board_data(10, board_shim)
println(data)
BrainFlow.release_session(board_shim)

File: python_package/examples/tests/brainflow_get_data_twice.py
import time

from brainflow.board_shim import BoardShim, BrainFlowInputParams, BoardIds
from brainflow.data_filter import DataFilter
from brainflow.ml_model import MLModel


def main():
    print('BoardShim version: ' + BoardShim.get_version())
    print('DataFilter version: ' + DataFilter.get_version())
    print('MLModel version: ' + MLModel.get_version())
    BoardShim.enable_dev_board_logger()

    params = BrainFlowInputParams()
    board = BoardShim(BoardIds.SYNTHETIC_BOARD.value, params)

    for i in range(2):
        board.prepare_session()
        board.start_stream()
        time.sleep(10)
        board.stop_stream()
        data = board.get_board_data()
        print(DataFilter.calc_stddev(data[2]))
        data = board.get_board_data()
        board.release_session()


if __name__ == "__main__":
    main()
